main.c

```diff
@@ -831,18 +893,18 @@ pile_from_solvables (Pool  *pool,
   queue_init (&sel);
   for (GStrv solvable = solvables; solvable && *solvable; solvable++)
     {
-      selection_make (pool, &sel, *solvable, sel_flags);
-      g_auto(Queue) q;
-      queue_init (&q);
-      selection_solvables (pool, &sel, &q);
-      if (!q.count)
+      /* solvables prefixed by @ are file names */
+      if (**solvable == '@')
+        add_solvables_from_file_to_pile (*solvable+1, pool, &sel, sel_flags, &pile, error);
+      else
+        add_solvable_to_pile (*solvable, pool, &sel, sel_flags, &pile);
+
+      /* if an error occurred, reset the pile and return */
+      if (*error)
         {
-          g_warning ("Nothing matches '%s'", *solvable);
-          continue;
+          queue_init (&pile);
```

Comment:
Shouldn't this be queue_free instead of queue_init?

Reply:
Since this function is returning a `Queue`, it's probably best not to free it before returning it. But it really doesn't matter since the caller will be exiting immediately because this is in the midst of returning an error.

Although, perhaps, this function should be adjusted so the caller instead passes in `pile` as a `Queue` pointer?